scheduler/service/service.go

```diff
@@ -379,112 +380,116 @@ func (s *Service) LeaveTask(ctx context.Context, req *rpcscheduler.PeerTarget) e
 	return nil
 }
 
-// StatPeerTask checks if the given task exists in P2P network
+// StatPeerTask checks the current state of the task
 func (s *Service) StatPeerTask(ctx context.Context, req *rpcscheduler.StatPeerTaskRequest) (*base.GrpcDfResult, error) {
-	log := logger.With("function", "StatPeerTask", "TaskID", req.TaskId)
-
-	// Check if task exists
 	task, loaded := s.resource.TaskManager().Load(req.TaskId)
-	if !loaded || task == nil {
-		msg := "task not found in P2P network"
-		log.Info(msg)
+	if !loaded {
+		msg := fmt.Sprintf("task %s not found", req.TaskId)
+		logger.Info(msg)
+		// TODO(eryugey) use dferrors.Newf
 		return common.NewGrpcDfResult(base.Code_PeerTaskNotFound, msg), nil
 	}
+
 	if task.FSM.Current() != resource.TaskStateSucceeded {
-		msg := fmt.Sprintf("task found but not in %s state: %s", resource.TaskStateSucceeded, task.FSM.Current())
-		log.Info(msg)
+		msg := fmt.Sprintf("task has been found but state is %s", task.FSM.Current())
+		task.Log.Info(msg)
+		// TODO(eryugey) use dferrors.Newf
 		return common.NewGrpcDfResult(base.Code_PeerTaskNotFound, msg), nil
 	}
 
-	lenPeers := task.PeerCount.Load()
-	if lenPeers <= 0 {
-		msg := fmt.Sprintf("task found but with %d active peers", lenPeers)
-		log.Info(msg)
+	if !task.HasAvailablePeer() {
+		msg := "task has been found but task has no available peers"
+		task.Log.Info(msg)
+		// TODO(eryugey) use dferrors.Newf
 		return common.NewGrpcDfResult(base.Code_PeerTaskNotFound, msg), nil
 	}
 
-	msg := "task found in P2P network"
-	log.Info(msg)
```

Comment:
Also I'd like a msg printed in log when the task is found, again, it might make debug life easier :)